fdbserver/include/fdbserver/IEncryptionKeyProvider.actor.h

```diff
@@ -253,22 +255,31 @@ class TenantAwareEncryptionKeyProvider : public IEncryptionKeyProvider {
 			}
 		}
 		if (domainId == SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID) {
-			*domainName = FDB_DEFAULT_ENCRYPT_DOMAIN_NAME;
+			*domainName = FDB_DEFAULT_ENCRYPT_DOMAIN_NAME_REF;
 		}
 		return domainId;
 	}
 
 	int64_t getTenant(const KeyRef& key, bool inclusive) {
 		// A valid tenant id is always a valid encrypt domain id.
 		static_assert(ENCRYPT_INVALID_DOMAIN_ID < 0);
-		if (key.size() < TENANT_PREFIX_SIZE || key >= systemKeys.begin) {
+
+		if (key.size() && key >= systemKeys.begin) {
 			return SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID;
 		}
-		// TODO(yiwu): Use TenantMapEntry::prefixToId() instead.
-		int64_t tenantId = bigEndian64(*reinterpret_cast<const int64_t*>(key.begin()));
-		if (tenantId < 0) {
-			return SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID;
+
+		if (key.size() < TENANT_PREFIX_SIZE) {
+			// Encryption domain information not available, leverage 'default encryption domain'
+			return FDB_DEFAULT_ENCRYPT_DOMAIN_ID;
+		}
+
+		StringRef prefix = key.substr(0, TENANT_PREFIX_SIZE);
+		int64_t tenantId = TenantMapEntry::prefixToId(prefix, false /* enforceValidTenantId */);
```

Comment:
Rather than use a comment to name the boolean, you can use our boolean parameters mechanism.

Reply:
Pardon my ignorance, what is `boolean parameter mechanism`? Are we referring to define `enum` to pass required information around.
Thanks

Reply:
I think `FDB_DECLARE_BOOLEAN_PARAM` is the one you were suggesting, will update the code accordingly. 
Thanks